rust/arrow/src/array/builder.rs

```diff
@@ -253,8 +253,18 @@ pub trait BufferBuilderTrait<T: ArrowPrimitiveType> {
 }
 
 impl<T: ArrowPrimitiveType> BufferBuilderTrait<T> for BufferBuilder<T> {
-    default fn new(capacity: usize) -> Self {
```

Comment:
Here, this implementation now takes care of both numerics and booleans, using `T::DATA_TYPE` to branch between them.